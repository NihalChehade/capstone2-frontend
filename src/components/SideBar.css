.navbar{
  background: rgba(161, 153, 153, 0.5) !important; 
  }

.nav-item {
    width: 130px;
    text-align: center;
  }

  nav a:not(.navbar-brand):hover {
    font-weight: bold;
  }
  