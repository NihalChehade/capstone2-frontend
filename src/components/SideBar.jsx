import React, { useState, useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../UserContext";
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
} from "reactstrap";
import "./SideBar.css";

function SideBar() {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);
  const { currentUser } = useContext(UserContext);

  return (
    <Navbar expand="md" className="navbar">
      <NavbarBrand href="/">Home Automation</NavbarBrand>
      <NavbarToggler onClick={toggle} />
      <Collapse isOpen={isOpen} navbar>
        <Nav className="mr-auto" navbar>
          <NavItem className="nav-item">
            <NavLink tag={Link} to="/" onClick={toggle}>
              {currentUser ? "My" : ""} Dashboard
            </NavLink>
          </NavItem>
          <NavItem className="nav-item">
            <NavLink tag={Link} to="/instructions" onClick={toggle}>
              Instructions
            </NavLink>
          </NavItem>
          {currentUser ? (
            <>
              <NavItem className="nav-item">
                <NavLink tag={Link} to="/devices/add" onClick={toggle}>
                  Add Device
                </NavLink>
              </NavItem>
              <NavItem className="nav-item">
                <NavLink tag={Link} to="/profile" onClick={toggle}>
                  Profile
                </NavLink>
              </NavItem>
              <NavItem className="nav-item">
                <NavLink tag={Link} to="/logout" onClick={toggle}>
                  Logout
                </NavLink>
              </NavItem>
            </>
          ) : (
            <>
              <NavItem className="nav-item">
                <NavLink tag={Link} to="/signup" onClick={toggle}>
                  Signup
                </NavLink>
              </NavItem>
              <NavItem className="nav-item">
                <NavLink tag={Link} to="/login" onClick={toggle}>
                  Login
                </NavLink>
              </NavItem>
            </>
          )}
        </Nav>
      </Collapse>
    </Navbar>
  );
}

export default SideBar;
