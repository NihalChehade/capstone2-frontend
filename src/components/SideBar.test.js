import React from 'react';
import { BrowserRouter} from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import SideBar from './SideBar';
import UserContext from '../UserContext';

describe('SideBar', () => {
  it('renders without crashing for guest users', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={{ currentUser: null }}>
          <SideBar />
        </UserContext.Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without crashing for logged-in users', () => {
    const userContextValueLoggedIn = {
      currentUser: { username: 'testuser' },
    };

    const { asFragment } = render(
      <BrowserRouter>
        <UserContext.Provider value={userContextValueLoggedIn}>
          <SideBar />
        </UserContext.Provider>
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('navigates to profile correctly', () => {
    const userContextValueLoggedIn = {
      currentUser: { username: 'testuser' },
    };

    render(
      <BrowserRouter>
        <UserContext.Provider value={userContextValueLoggedIn}>
          <SideBar />
        </UserContext.Provider>
      </BrowserRouter>
    );

    // Simulate a click on the "Profile" link
    fireEvent.click(screen.getByText(/Profile/i));
    // Check if the URL changes to "/profile" after clicking
    expect(window.location.pathname).toEqual('/profile');
  });
});
