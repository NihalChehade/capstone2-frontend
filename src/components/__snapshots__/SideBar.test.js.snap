// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders without crashing for guest users 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar navbar-expand-md"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Home Automation
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="mr-auto navbar-nav"
        >
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/"
            >
               Dashboard
            </a>
          </li>
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/instructions"
            >
              Instructions
            </a>
          </li>
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/signup"
            >
              Signup
            </a>
          </li>
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/login"
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`SideBar renders without crashing for logged-in users 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar navbar-expand-md"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        Home Automation
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="mr-auto navbar-nav"
        >
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/"
            >
              My Dashboard
            </a>
          </li>
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/instructions"
            >
              Instructions
            </a>
          </li>
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/devices/add"
            >
              Add Device
            </a>
          </li>
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item nav-item"
          >
            <a
              class="nav-link"
              data-discover="true"
              href="/logout"
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
